synapse/handlers/receipts.py

```diff
@@ -178,35 +179,27 @@ def filter_out_hidden(events: List[JsonDict], user_id: str) -> List[JsonDict]:
 
             for event_id in content.keys():
                 event_content = content.get(event_id, {})
-                m_read = event_content.get(ReceiptTypes.READ, {})
 
-                # If m_read is missing copy over the original event_content as there is nothing to process here
-                if not m_read:
-                    new_event["content"][event_id] = event_content.copy()
+                m_read = event_content.get(ReceiptTypes.READ, None)
+                if m_read:
+                    new_event["content"][event_id] = {ReceiptTypes.READ: m_read}
                     continue
 
-                new_users = {}
-                for rr_user_id, user_rr in m_read.items():
-                    try:
-                        hidden = user_rr.get("hidden")
-                    except AttributeError:
-                        # Due to https://github.com/matrix-org/synapse/issues/10376
-                        # there are cases where user_rr is a string, in those cases
-                        # we just ignore the read receipt
-                        continue
+                m_read_private = event_content.get(ReceiptTypes.READ_PRIVATE, None)
+                if m_read_private:
+                    new_users = {}
+                    for rr_user_id, user_rr in m_read_private.items():
+                        if rr_user_id == user_id:
+                            new_users[rr_user_id] = user_rr.copy()
+
+                    # Set new users unless empty
+                    if len(new_users.keys()) > 0:
+                        new_event["content"][event_id] = {
+                            ReceiptTypes.READ_PRIVATE: new_users
+                        }
+                    continue
```

Comment:
I think you can avoid the loop and heavily simplify this:

```suggestion
                    user_rr = m_read_private.get(user_id)
                    if user_rr:
                        new_event["content"][event_id] = {
                            ReceiptTypes.READ_PRIVATE: {user_id: user_rr.copy()}
                        }
```

Although I think the current code (and my code above) will have issues if the current user provides both a private and non-private read receipt for the same event. This is a silly thing to do, but I suspect we should send both back?